service/dynamodb/dynamodbattribute/decode.go

```diff
@@ -220,7 +220,7 @@ func (d *Decoder) decodeBinary(b []byte, v reflect.Value) error {
 	switch v.Type().Elem().Kind() {
 	case reflect.Uint8:
 		// Fallback to reflection copy for type aliased of []byte type
-		if v.IsNil() || v.Cap() < len(b) {
+		if (v.Kind() != reflect.Array && v.IsNil()) || v.Cap() < len(b) {
 			v.Set(reflect.MakeSlice(v.Type(), len(b), len(b)))
```

Comment:
I might be reading this wrong, but this seems that this case would be triggered if V is an Array, and Cap is less than `len(b)`.  I think this would trigger a panic by trying to set a slice to array value.